app/models/good_job/execution_result.rb

```diff
@@ -9,17 +9,24 @@ class ExecutionResult
     # @return [Exception, nil]
     attr_reader :unhandled_error
     # @return [Exception, nil]
+    attr_reader :unlocked_error
+    # @return [Exception, nil]
     attr_reader :retried
     alias retried? retried
 
     # @param value [Object, nil]
     # @param handled_error [Exception, nil]
     # @param unhandled_error [Exception, nil]
-    def initialize(value:, handled_error: nil, unhandled_error: nil, retried: false)
+    def initialize(value:, handled_error: nil, unhandled_error: nil, unlocked_error: nil, retried: false)
```

Comment:
just curious, but why not just case on the type of the error?

Reply:
oh! `unlocked_error` (I just named it to `unexecutable`) and `retried` are both boolean flags. It was in the spirit of what's in #995.

Just to explain, this is all a bit messy because it involves extracting state out of Active Job by storing it in a thread-local variable and then using it later.